File: src/main/resources/dbconnector.js
/**
 * Create a connection to a database
 * @param connString ConnectionString to connect to a database type
 * @param username login name of the database
 * @param password login password of the database
 * @constructor init a database connection
 */
var DbConnector = function(connString, username, password) {
    try {
        var DBConnector = Java.type("nl.mawoo.wcmscript.extend.dbconnector.DbConnector");
        this.connector = new DBConnector(connString, username, password);
    } catch(e) {
        print("Could not connect to the database \n Check your connection settings.");
    }
};

/**
 * Run a query to the database.
 * @param sql database query.
 */
DbConnector.prototype.query = function(sql){
    try {
        return this.connector.query(sql);
    } catch(e) {
        print("A SQL error occurred: "+ e);
    }
};

File: src/main/java/nl/mawoo/wcmscript/extend/dbconnector/DbConnector.java
package nl.mawoo.wcmscript.extend.dbconnector;

import nl.mawoo.wcmscript.logger.AbstractLogger;
import nl.mawoo.wcmscript.logger.WCMSLogger;
import org.apache.log4j.Logger;

import java.sql.*;

/**
 * This class is responsible to connect to a database.
 *
 * @author Bob van der Valk
 */
public class DbConnector {
    static final String JDBC_DRIVER = "com.mysql.jdbc.Driver";
    private AbstractLogger logger = WCMSLogger.getLogger(DbConnector.class);

    private Connection conn = null;
    private Statement stmt = null;


    /**
     * Set up a database connection with JDBC.
     * @param connection string for your connection
     * @param username user of database
     * @param password password of database
     * @throws ClassNotFoundException
     */
    public DbConnector(String connection, String username, String password) throws ClassNotFoundException {
        Class.forName("com.mysql.jdbc.Driver");

        try {
            conn = DriverManager.getConnection(connection, username, password);
        } catch (SQLException e) {
            logger.error("Cannot connect to database: "+ e);
        }
    }

    /**
     * Run a sql query and get results back.
     * @param sql your sql manager
     * @return resultSet of your query
     */
    public ResultSetObject query(String sql) {
        try {
            stmt = conn.createStatement();
            String[] queryType = sql.split(" ");

            if("SELECT".equals(queryType[0])) {
                ResultSet rs = stmt.executeQuery(sql);

                return new ResultSetObject(rs);
            } else {
                stmt.execute(sql);
            }
        } catch (SQLException e) {
            logger.error("A SQL exception occurred: "+ e);
        }

        return null;
    }
}
